src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public static class SyntaxNodeExtensions
+    {
+        internal static Range GetRange(this SyntaxNode syntaxNode, RazorCodeDocument codeDocument)
+        {
+            try
+            {
+                int startPosition;
+                int endPosition;
+                if (syntaxNode is MarkupTagHelperAttributeSyntax thAttributeSyntax)
```

Comment:
It doesn't feel right to special case these at this layer. Instead just call `GetRange()` on the `Name` node directly if you want the Range just for the name.